### PR TITLE
[Part of #69] AI Players

### PR DESCRIPTION
**Pull request by elbrujohalcon**
[Part of #69] AI Players

### CODE REVIEW

File: src/clients/ai/spts_cli.erl [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+             , serpent_id = SerpentId
+             , game       = GameDesc
+             , mod        = Module
+             , mod_state  = ModState
+             }}.
+
+-spec handle_cast(stop, state()) -> {stop, normal, state()}.
+handle_cast(stop, State) -> {stop, normal, State}.
+
+-spec handle_info(term(), state()) -> {noreply, state()} | {stop, normal, state()}.
```
Comment (gadgetci) [inline src/clients/ai/spts_cli.erl]: According to **Elvis**:
> Line 79 is too long: -spec handle_info(term(), state()) -> {noreply, state()} | {stop, normal, state()}..

File: src/protocol/spts_hdp_game_handler.erl [hunk 2]
```diff
@@ -211,6 +211,7 @@ update_all_users(State) ->
       cxy_ctl:execute_task(spts_hdp, ?MODULE, update_user, [User, State])
     end, Users).
 
+-spec update_user(#user{}, state()) -> ok.
```
Comment (gadgetci) [inline src/protocol/spts_hdp_game_handler.erl]: According to **Elvis**:
> The spec in line 214 uses a record, please define a type for the record and use that instead.

Comment (gadgetci) [inline src/protocol/spts_hdp_game_handler.erl]: According to **Elvis**:
> The spec in line 214 uses a record, please define a type for the record and use that instead.

File: src/clients/ai/spts_cli_random.erl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Callback implementation
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-spec init(pos_integer(), spts_hdp:game(), noargs) -> state().
+init(_SerpentId, _Game, noargs) -> #state{}.
+
+-spec handle_update(
+  [spts_hdp:diff()], pos_integer(), spts_hdp:game(), state()) ->
+  {none | spts_games:direction(), state()}.
+handle_update([], _SerpentId, _Game, State) ->
```
Comment (gadgetci) [inline src/clients/ai/spts_cli_random.erl]: According to **Xref**:
> `spts_cli_random:handle_update/4` calls undefined function `ktn_random:pick/1`

File: src/clients/ai/spts_cli_random.erl [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  %NOTE: no changes, nothing to do
+  {none, State};
+handle_update(Diffs, SerpentId, Game, State) ->
+  lager:notice(
+    "\n\tDiffs = ~p,\n\tSerpentId = ~p,\n\tGame = ~p",
+    [Diffs, SerpentId, Game]),
+  Direction = ktn_random:pick([up, down, left, right]),
```
Comment (gadgetci) [inline src/clients/ai/spts_cli_random.erl]: According to **Dialyzer**:
>  Call to missing or unexported function ktn_random:pick/1

Comment (gadgetci) [inline src/clients/ai/spts_cli_random.erl]: According to **Dialyzer**:
>  Call to missing or unexported function ktn_random:pick/1

**Comment by gadgetci** [comment]
According to **Dialyzer**:
> 
dialyzer: Unknown option: ebin/

File: Makefile [hunk 5]
```diff
@@ -19,6 +19,7 @@ dep_xref_runner = git https://github.com/inaka/xref_runner.git 0.2.2
 dep_recon = git https://github.com/ferd/recon.git 2.2.1
 dep_epocxy = git https://github.com/duomark/epocxy.git 0.9.9
```
Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> /tmp/gadget/inaka/serpents/0001440-0105370-0943700/erlang.mk: No such file or directory

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> /tmp/gadget/inaka/serpents/0001440-0105370-0943700/erlang.mk: No such file or directory

File: src/clients/ai/spts_cli.erl (hunk 1 above)
Comment (gadgetci) [inline src/clients/ai/spts_cli.erl]: According to **Elvis**:
> Line 79 is too long: -spec handle_info(term(), state()) -> {noreply, state()} | {stop, normal, state()}..

**Comment by gadgetci** [comment]
According to **Dialyzer**:
> 
dialyzer: Unknown option: ebin/

File: Makefile (hunk 5 above)
Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> /tmp/gadget/inaka/serpents/0001440-0105370-0943700/erlang.mk: No such file or directory

**Comment by gadgetci** [comment]
According to **Dialyzer**:
> 
dialyzer: Unknown option: ebin/

File: Makefile [hunk 6]
```diff
@@ -9,7 +9,7 @@ SELL_DEPS = sync
 TEST_DEPS = xref_runner shotgun
 
 dep_lasse = git https://github.com/inaka/lasse.git 1.0.1
-dep_katana = git https://github.com/inaka/erlang-katana.git 0.2.7
+dep_katana = git https://github.com/inaka/erlang-katana.git 0.2.9
 dep_cowboy = git https://github.com/extend/cowboy.git 1.0.1
 dep_jiffy = git https://github.com/davisp/jiffy.git 0.13.3
 dep_mixer = git https://github.com/inaka/mixer.git 0.1.2
```
Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

File: src/utils/spts_hdp.erl [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+-spec game(pos_integer(), pos_integer()) -> binary().
+game(MsgId, GameId) ->
+  H = head(2, MsgId),
+  <<H/binary, GameId:?USHORT>>.
+
+-spec games(pos_integer()) -> binary().
+games(MsgId) -> head(2, MsgId).
+
+-spec ping(pos_integer()) -> binary().
+ping(MsgId) -> head(1, MsgId).
```
Comment (HernanRivasAcosta) [inline src/utils/spts_hdp.erl]: Not rejecting a PR because of it, but this magic numbers feel off.

Comment (elbrujohalcon) [inline src/utils/spts_hdp.erl]: Ooops!

File: Makefile (hunk 6 above)
Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

Comment (gadgetci) [inline Makefile]: According to **Compiler**:
> warning: overriding commands for target `shell'

File: src/utils/spts_hdp.erl [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+head(Flags, MsgId, UserTime, UserId) ->
+  <<Flags:?UCHAR, MsgId:?UINT, UserTime:?USHORT, UserId:?UINT>>.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%% Internal Functions
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+msg_type(ping) -> 1;
+msg_type(info) -> 2;
+msg_type(join) -> 3;
+msg_type(client_update) -> 4.
```
Comment (HernanRivasAcosta) [inline src/utils/spts_hdp.erl]:
``` erlang
msg_type(Other) -> Other.
```

Comment (elbrujohalcon) [inline src/utils/spts_hdp.erl]: Just message types… it's fine.

**Pull request merged**